infra/src/main/kotlin/fi/fta/geoviite/infra/linking/Linking.kt

```diff
@@ -86,6 +86,12 @@ data class SuggestedSwitch(
     val alignmentEndPoint: LocationTrackEndpoint?,
     val geometryPlanId: IntId<GeometryPlan>? = null,
     val geometrySwitchId: IntId<GeometrySwitch>? = null,
+    val topologicalJointConnections: List<TopologicalJointConnection>? = null,
+)
+
+data class TopologicalJointConnection(
+    val jointNumber: JointNumber,
+    val locationTrackIds: List<IntId<LocationTrack>>,
```

Comment:
Samankaltainen tietorakenne kuin `SuggestedSwitchJoint`. 

Voipi olla, että nämä kannattaisi yhtenäistää, mutta pidetty toistaiseksi muutosmäärän rajaamiseksi erillisinä. 

Yleishuomionahan tämä koodikanta on aika täynnä näitä listoja: Tässäkin oma intuitio sanoisi, että `Map<JointNumber, List<IntId<LocationTrack>>` olisi kuvaavampi^, koska sellaistahan rakennetta tässä haetaan. Toki tuo mappien käyttäminen taas TS:n puolella on rasittavampaa, kun esimerkiksi `Object.entries()/.values()/yms` eivät ylläpidä tyyppi-informaatiota.

---

^Suuremmissa tietorakenteissa myös perffin puolesta oletettavasti nopeampi, kun ei jatkuvasti tarvitse käydä listaa läpi esim. filtteröimällä tiettyyn idseen)

Reply:
Frontille palautettaessa suosisin itsekin ennemmin kuvaavista tietotyypeistä tehtyjä listoja kuin mappeja, sillä mapit serialisoituvat geneerisiksi listoiksi joissa sisältönä on `{ key: <avain>, value: <arvo> }`. That being said, jos jostain muusta muodosta on hyötyä bäkkärillä sisäisesti, niin tämän mäppäyksen voi aina tehdä vaikka vasta Controllerissa.